src/librustdoc/html/markdown.rs

```diff
@@ -91,15 +91,15 @@ impl<'a> Line<'a> {
 fn map_line(s: &str) -> Line {
     let trimmed = s.trim();
     if trimmed.starts_with("##") {
-        Line::Shown(&trimmed[1..])
+        Line::Shown(Cow::Owned(s.replacen("##", "#", 1)))
```

Comment:
Two things:

1. Wait, there is not Cow-returning replacen yet? Might be a nice thing to have, [since it's pretty concise](https://github.com/rust-lang/rust/blob/4ecf12bf0eb8386626ccdb5f721a7183ccc4eba6/src/liballoc/str.rs#L305-L316). Nit relevant here.
2. Given that we are in the branch where the string starts with `##`, isn't replacing the two hash signs with one the same as `&x[1..]`?

Reply:
1. Agreed.
2. It is _not_ the same, because of leading whitespace. We check if `trimmed` starts with `##`, not `s`.

Reply:
Oh right, my bad